docs/dev_cycle.part-03.rst

```diff
@@ -0,0 +1,194 @@
+
+.. contents:: :local:
+
+Tests and Chains
+----------------
+
+For testing, we will use the ``tester`` chain again. It's very convinient blockchain for tests,
+because it reset on each run, and the state is saved only in memory and cleared after each run.
+In a way, this is a similar idea to running tests against a DB,
+where you create an ad-hoc temporary DB for the tests. 
+
+You will run Populus tests with ``py.test``, which was installed when you installed Populus.
+
+Add a test file:
+
+.. code-block:: bash
+
+  $ nano tests/test_donator.py
+  
+.. note::
+  
+    py.test collects all the tests that follow
+    it's `naming conventions <https://pytest.readthedocs.io/en/reorganize-docs/new-docs/user/naming_conventions.html>`_
+    
+We don't need the Greeter example for this project, so delete it:
+
+.. code-block:: bash
+
+  $ rm contracts/Greeter 
+  4 rm tests/test_greeter.py
+  
+Now, before we start writing tests, pause for a moment: What are we actually testing? Obviously a contract, but what *is* a contract?
+
+
+What is a Contract?
+-------------------
+
+The simplest definition of a contract is a compiled program that runs on the Ethereum blockchain.
+It's a bytecode of instructions, saved on the blockchain, that a blockchain node can run. 
+The node gives this program a sandbox, a closed environment, to execute. 
+
+In a way, if the blockchain is the OS, then a contract is an executable that runs by the OS. 
+Everything else: syncing nodes, consensus, mining, is OS stuff. Yes, it's a smart
+OS that magically syncs a lot of nodes, but the contract doesn't care - it just runs, and is allowed to use the API that the OS
+gives it.
+
+However, the term *contract* can be confusing, since it's used for several different things:
+
+#. A bytecode program that is saved on the blockchain
+#. A ``contract`` code in a Solidity source file
+#. A Web3 ``contract`` object
+
+Luckily, it can be even more confusing. You may have a Solidity source file of the ``Donator`` contract. If you deployed it 
+to ``testnet``, then we have another "contract", the code that sits on ``testnet``. When you deploy to ``mainnet``,
+which is another blockchain, we now have three contracts: the source file, the bytecode on ``testnet``, and the bytecode on ``mainnet``.
+
+But wait, there is more! To interact with an *existing* contract on say ``mainnet``, we need a Web3 "contract" *object*. This object does
+not need the solidity source, since the bytcode is alreadt compiled and deployed. It does need the ABI,  the functions 
+and arguments structure of the *bytecode* contract's interface, and the address of this *bytecode* contract
+on the mainnet.
+
+.. note::
+  To deploy a *new* ``Donator`` contract, maybe to another blockchain, or to another address on the same blockchain,
+  we also need a Web3 contract *object*, but this time the object does require a compiled
+  "contract"
+
+Actually when we say ""contract", we deal with three different things:
```

Comment:
This sentence is awkward.  I feel like this list and information is already present a few lines up.

Reply:
ok, summery in one sent